arraycontext/impl/pytato/compile.py

```diff
@@ -550,13 +550,15 @@ def _args_to_device_buffers(actx, input_id_to_name_in_program, arg_id_to_arg):
             pass
         elif isinstance(arg, pt.Array):
             # got an array expression => evaluate it
-            from warnings import warn
-            warn(f"Argument array '{arg_id}' to a compiled function is "
-                    "unevaluated. Evaluating just-in-time, at "
-                    "considerable expense. This is deprecated and will stop "
-                    "working in 2023. To avoid this warning, force evaluation "
-                    "of all arguments via freeze/thaw.",
-                    DeprecationWarning, stacklevel=4)
+            from warnings import catch_warnings, filterwarnings, warn
+            with catch_warnings():
+                filterwarnings("always")
+                warn(f"Argument array '{arg_id}' to the '{fn_name}' compiled function "
+                     "is unevaluated. Evaluating just-in-time, at "
+                     "considerable expense. This is deprecated and will stop "
+                     "working in 2023. To avoid this warning, force evaluation "
+                     "of all arguments via freeze/thaw.",
+                      stacklevel=4)
```

Comment:
I think it'd be useful to create a special warning class for this:
```
class UnevaluatedCompiledFuncArgWarning(UserWarning):
    pass
```
(and then pass the appropriate argument to `warn`) This makes precise filtering from the application side possible. I don't know that I want to install a warning filter on the user's behalf. (We could, in the actx constructor, on an opt-out basis? But that still feels overbearing.)

Alternatively, we could just make this an error, given that it has been deprecated with a deadline that has passed.

Reply:
I'm starting to like the "make it an error" idea, I think.

Reply:
> I'm starting to like the "make it an error" idea, I think.

Same here :)

Reply:
Done in [5535398](https://github.com/inducer/arraycontext/pull/305/commits/553539813f442f85cf6c92765f8854dc1c0b5b8e)